sem/logging.py

```diff
@@ -92,6 +94,7 @@ def parse_logs(log_file):
     # Note: '^$' - Ensures that the entire regex matches the entire log line.
     regex = re.compile(r'^' + time_re + context_extended_context_re + component_function_arguments_re + severity_class_message_re + r'$')
 
+    idx = 0
     with open(log_file) as f:
         for log in f:
```

Comment:
one of python's best practices is to avoid creating manual counters when using loops.
```suggestion
        for idx, log in enumerate(f):
```

Reply:
I initially thought of this but we have to increment the index only if the log line is parsed correctly. If the control goes here
`if groups is None:
warnings.warn("Log format is not consistent with prefix_all. Skipping log '%s'" % log, RuntimeWarning, stacklevel=2)
continue `

we do not increment the index.

Reply:
Okay, i get it now, thank you for the explanation.
Though, I would suggest the opposite: in this way row number and log index would be consistent, and a user would easily be able to find any given line if he needs to.
On the other hand, having the index and the row number as possibly different might be annoying for a user.

What do you guys think?

Reply:
I did not quite understand what approach did you referred to as the opposite. Can you please elaborate a little? Also, I would like to just point out that the index that we store in the database is not visible to the user at any point. It is just to make searching for the log entries easier and faster(as the index are always sorted).